ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -1610,6 +1644,12 @@ private static Map<String, ConfigValue> buildStreamingConfig(
     final Map<String, ConfigValue> streamConfigProps = new HashMap<>();
     applyStreamsConfig(baseStreamConfig, streamConfigProps);
     applyStreamsConfig(overrides, streamConfigProps);
+
+    // Streams client metrics aren't used in Confluent deployment
+    if (streamConfigProps.get(KSQL_DEPLOYMENT_TYPE_CONFIG).value.equals(DeploymentType.confluent.name())) {
+      streamConfigProps.entrySet().stream().filter(e -> isKeyPrefixed(e.getKey(), TELEMETRY_PREFIX)).forEach(streamConfigProps::remove);
```

Comment:
I was thinking this as well but I thought this approach was safer because I was worried that there may be other metric reporters that may need to get configured. WDYT?